### PR TITLE
[SolYul] Division.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

**Review by leonardoalt — COMMENTED** [review]
Looks good, hard to say at a first glance why that test is failing

**Comment by chriseth** [comment]
Test is failing because there is an overflow. Will remove it again.

**Review by leonardoalt — DISMISSED** [review]

**Comment by chriseth** [comment]
Rebased.

File: libsolidity/codegen/YulUtilFunctions.cpp [hunk 1]
```diff
@@ -374,6 +374,33 @@ string YulUtilFunctions::overflowCheckedUIntMulFunction(size_t _bits)
...
+	return m_functionCollector->createFunction(functionName, [&]() {
+		return
+			Whiskers(R"(
+			function <functionName>(x, y) -> r {
+				if iszero(y) { revert(0, 0) }
+				<?signed>
+				// x / -1 == x
+				if and(
+					eq(y, sub(0, 1)),
+					eq(x, <minVal>)
```
Comment (Marenz) [inline libsolidity/codegen/YulUtilFunctions.cpp]: I would swap those two `eq` lines because you usually expect x first given the equation `x / y`

**Comment by chriseth** [comment]
Updated.

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**